lib/ansible/runner/credential.rb

```diff
@@ -0,0 +1,54 @@
+module Ansible
+  class Runner
+    class Credential
+      attr_reader :auth, :base_dir
+
+      def self.new(authentication_id, base_dir)
+        auth_type = Authentication.find(authentication_id).type
+        self == Ansible::Runner::Credential ? detect_credential_type(auth_type).new(authentication_id, base_dir) : super
+      end
+
+      def self.detect_credential_type(auth_type)
+        subclasses.index_by(&:auth_type)[auth_type] || Ansible::Runner::GenericCredential
+      end
+
+      def initialize(authentication_id, base_dir)
+        @auth     = Authentication.find(authentication_id)
+        @base_dir = base_dir
+
+        FileUtils.mkdir_p(env_dir)
```

Comment:
Make sure this isn't world readable.

Reply:
I think the base dir is created as `0700` based on https://ruby-doc.org/stdlib-2.5.3/libdoc/tmpdir/rdoc/Dir.html#method-c-mktmpdir so I think we're good there.

Reply:
Yay!

(as if I didn't know this already because we were just chatting about this in person 😉 )